backend/Lingva.BusinessLayer/SubtitlesParser/Classes/Parsers/SsaParser.cs

```diff
@@ -0,0 +1,172 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+
+namespace Lingva.BusinessLayer.SubtitlesParser.Classes.Parsers
+{
+    /// <summary>
+    /// A parser for the SubStation Alpha subtitles format.
+    /// See http://en.wikipedia.org/wiki/SubStation_Alpha for complete explanations.
+    /// Ex:
+    /// [Script Info]
+    /// ; This is a Sub Station Alpha v4 script.
+    /// ; For Sub Station Alpha info and downloads,
+    /// ; go to http://www.eswat.demon.co.uk/
+    /// Title: Neon Genesis Evangelion - Episode 26 (neutral Spanish)
+    /// Original Script: RoRo
+    /// Script Updated By: version 2.8.01
+    /// ScriptType: v4.00
+    /// Collisions: Normal
+    /// PlayResY: 600
+    /// PlayDepth: 0
+    /// Timer: 100,0000
+    ///  
+    /// [V4 Styles]
+    /// Format: Name, Fontname, Fontsize, PrimaryColour, SecondaryColour, TertiaryColour, BackColour, Bold, Italic, BorderStyle, Outline, Shadow, Alignment, MarginL, MarginR, MarginV, AlphaLevel, Encoding
+    /// Style: DefaultVCD, Arial,28,11861244,11861244,11861244,-2147483640,-1,0,1,1,2,2,30,30,30,0,0
+    ///   
+    /// [Events]
+    /// Format: Marked, Start, End, Style, Name, MarginL, MarginR, MarginV, Effect, Text
+    /// Dialogue: Marked=0,0:00:01.18,0:00:06.85,DefaultVCD, NTP,0000,0000,0000,,{\pos(400,570)}Like an angel with pity on nobody
+    /// </summary>
+    public class SsaParser : ISubtitlesParser
+    {
+        // Properties ---------------------------------------------------------------
+
+        private const string EventLine = "[Events]";
+        private const char Separator = ',';
+
+        private const string StartColumn = "Start";
+        private const string EndColumn = "End";
+        private const string TextColumn = "Text";
+
+
+        // Methods ------------------------------------------------------------------
+
+        public List<SubtitleItem> ParseStream(Stream ssaStream, Encoding encoding)
```

Comment:
Please, refactor this method to make it concise and contains less if-else nested blocks